x-pack/plugins/security_solution/common/detection_engine/schemas/types/threat_mapping.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/* eslint-disable @typescript-eslint/naming-convention */
+
+import * as t from 'io-ts';
+
+export const threat_query = t.string;
+export type ThreatQuery = t.TypeOf<typeof threat_query>;
+export const threatQueryOrUndefined = t.union([threat_query, t.undefined]);
+export type ThreatQueryOrUndefined = t.TypeOf<typeof threatQueryOrUndefined>;
+
+export const threat_filters = t.array(t.unknown); // Filters are not easily type-able yet
+export type ThreatFilters = t.TypeOf<typeof threat_filters>;
+export const threatFiltersOrUndefined = t.union([threat_filters, t.undefined]);
+export type ThreatFiltersOrUndefined = t.TypeOf<typeof threatFiltersOrUndefined>;
+
+export const threatMappingEntries = t.array(
+  t.exact(
+    t.type({
+      field: t.string,
```

Comment:
What do you think about making this the non-empty string type? Just remembering with exceptions, we had to update it to that after the fact to avoid some bugs that were happening. Not sure if that's of concern here.

Reply:
Agreed, I think that is a good idea. Changed the code now to reflect this. You will see it on the next commit.